carbonmark/components/pages/Retire/RetireModal/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { Text } from "@klimadao/lib/components";
+import { Trans } from "@lingui/macro";
+import { Spinner } from "components/shared/Spinner";
+import { Transaction } from "components/Transaction";
+import { TransactionStatusMessage } from "lib/statusMessage";
+import { CarbonmarkToken } from "lib/types/carbonmark";
+import { StaticImageData } from "next/image";
+import { FC, ReactNode, useEffect, useState } from "react";
+import { CustomizableModal } from "../CustomizableModal";
+import * as styles from "./styles";
+
+interface Props {
+  title: ReactNode;
+  value: string;
+  approvalValue: string;
+  token: {
+    key: string;
+    icon: StaticImageData;
+    label: Uppercase<CarbonmarkToken>;
+  };
+  spenderAddress: string;
+  onCloseModal: () => void;
+  onApproval: () => void;
+  onSubmit: () => void;
+  status: TransactionStatusMessage | null;
+  setStatus: (status: TransactionStatusMessage | null) => void;
+  onResetStatus: () => void;
+  isApproved: boolean;
+  showModal: boolean;
+}
+
+export const RetireModal: FC<Props> = (props) => {
+  const [txnView, setTxnView] = useState<"approve" | "submit">(
+    props.isApproved ? "submit" : "approve"
+  );
+
+  const [processingRetirement, setProcessingRetirement] = useState(false);
+
+  const statusType = props.status?.statusType;
+
+  useEffect(() => {
+    setTxnView(props.isApproved ? "submit" : "approve");
+  }, [props.isApproved]);
+
+  useEffect(() => {
+    if (statusType === "networkConfirmation" && txnView === "submit") {
+      setProcessingRetirement(true);
+    }
+  }, [props.status, txnView]);
```

Comment:
Sounds good, I actually lifted processingRetirement up one component as well. It triggers the processing modal as soon as metamask opens instead of on submit, which I think makes sense. Catches the error in handleRetire to set it to false.